apps/web/src/pages/Applications/ApplicationProfilePage/ApplicationProfilePage.tsx

```diff
@@ -75,16 +99,194 @@ export const getPageInfo: GetApplicationPageInfo = ({
   };
 };
 
-const ApplicationProfile = ({ application }: ApplicationPageProps) => {
+interface ApplicationProfileProps extends ApplicationPageProps {
+  user: User;
+}
+
+export const ApplicationProfile = ({
+  application,
+  user,
+}: ApplicationProfileProps) => {
   const intl = useIntl();
   const paths = useRoutes();
+  const navigate = useNavigate();
+  const { applicantDashboard } = useFeatureFlags();
   const pageInfo = getPageInfo({ intl, paths, application });
+  const [{ fetching: submitting }, executeSubmitMutation] =
+    useUpdateApplicationMutation();
+  const [{ fetching: isUpdating }, executeUpdateMutation] =
+    useUpdateUserAsUserMutation();
+  const nextStepPath = paths.applicationResumeIntro(application.id);
+
+  const handleNavigation = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault(); // We don't want to navigate until we mark the step as complete
+    const isValid =
+      application?.poolAdvertisement &&
+      !pageInfo?.hasError?.(user as Applicant, application?.poolAdvertisement);
+
+    if (isValid) {
+      executeSubmitMutation({
+        id: application.id,
+        application: {
+          insertSubmittedStep: ApplicationStep.Welcome,
```

Comment:
Welcome step was prior, review your profile is the one to add here right?

Reply:
Yeah, that makes sense. Not sure what I was thinking submitting the previous step 😕 lol

Fixed in [update to submit proper step](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6296/commits/5d64b23f57af3dd362ca3b980892ac1d60177062)